clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -154,6 +155,38 @@ public class AzureComputeClient extends AzureBaseClient {
     null
   }
 
+  ServiceResponse<Void> destroyServerGroup(String resourceGroupName, String serverGroupName) {
+    ServiceResponse<Void> result = null
+    // The API call might return a timeout exception or some other Azure CloudException that is not
+    //   related to the operation we are trying to execute; retry couple of times and if the final
+    //   retry fails then rethrow
+    long operationRetry = 0
+    while (operationRetry < AZURE_ATOMICOPERATION_RETRY) {
```

Comment:
@ajordens I moved most of the duplicated code that handles the retries into the base class; the two methods will retry on both getting the operation object and the actual delete op. 
In time I will continue to build on top of that and change some more operations to use the same pattern (see get like methods).